src/SharedWeb/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -693,16 +695,33 @@ public static void AddWebAuthn(this IServiceCollection services, GlobalSettings
         this IServiceCollection services,
         GlobalSettings globalSettings)
     {
-        if (globalSettings.SelfHosted || string.IsNullOrEmpty(globalSettings.Redis.ConnectionString))
+        if (!string.IsNullOrEmpty(globalSettings.DistributedCache?.Redis?.ConnectionString))
+        {
+            services.AddStackExchangeRedisCache(options =>
+            {
+                options.Configuration = globalSettings.DistributedCache.Redis.ConnectionString;
+            });
+        }
+        else
         {
             services.AddDistributedMemoryCache();
-            return;
         }
 
-        services.AddStackExchangeRedisCache(options =>
+        if (!string.IsNullOrEmpty(globalSettings.DistributedCache?.Cosmos?.ConnectionString))
         {
-            options.Configuration = globalSettings.Redis.ConnectionString;
-        });
+            services.AddKeyedSingleton<IDistributedCache>("persistent", (s, _) =>
+                new CosmosCache(new CosmosCacheOptions
+                {
+                    DatabaseName = "cache",
+                    ContainerName = "default",
```

Comment:
🌱 I want us to keep thinking about this and its planning for ideal expansion of use cases before this merges.

Reply:
I think this structure allows for expansion by implementing additional keyed services for IDistributedCache if we need/want to. Each of these could have their own database and/or container.

Reply:
Something about these names is too ... straightforward / default (quite literally) for me, but I guess I don't have a ... default alternative.